### PR TITLE
re: Fix remaining `clippy` warnings in `intergration-tests`

### PR DESCRIPTION
**Pull request by pmnoxx**
There are still a few `clippy` warnings that need to be fixed in `integration-tests`.

### CODE REVIEW

File: integration-tests/src/user/runtime_user.rs [hunk 1]
```diff
@@ -169,15 +169,14 @@ impl RuntimeUser {
             block_hash: Default::default(),
         }];
         for hash in &receipt_ids {
-            transactions
-                .extend(self.get_recursive_transaction_results(&hash.clone().into()).into_iter());
+            transactions.extend(self.get_recursive_transaction_results(&(*hash)).into_iter());
```
Comment (mina86) [inline integration-tests/src/user/runtime_user.rs:172]: `&*hash` should just be `hash`.

File: tools/restored-receipts-verifier/src/main.rs [hunk 2]
```diff
@@ -14,7 +14,7 @@ use nearcore::{get_default_home, get_store_path, load_config, NightshadeRuntime,
 fn get_receipt_hashes_in_repo() -> Vec<CryptoHash> {
     let receipt_result = load_migration_data(&"mainnet".to_string()).restored_receipts;
     let receipts = receipt_result.get(&0u64).unwrap();
-    receipts.into_iter().map(|receipt| receipt.get_hash()).collect()
+    receipts.iter().map(|receipt| receipt.get_hash()).collect()
```
Comment (mina86) [inline tools/restored-receipts-verifier/src/main.rs:17]: Why is `into_iter` an issue here?

**Review by miraclx — APPROVED** [review]

File: tools/restored-receipts-verifier/src/main.rs (hunk 2 above)
Comment (pmnoxx) [reply tools/restored-receipts-verifier/src/main.rs:17]:
```
warning: this `.into_iter()` call is equivalent to `.iter()` and will not consume the `Vec`
  --> tools/restored-receipts-verifier/src/main.rs:17:14
   |
17 |     receipts.into_iter().map(|receipt| receipt.get_hash()).collect()
   |              ^^^^^^^^^ help: call directly: `iter`
   |
   = note: `-W clippy::into-iter-on-ref` implied by `-W clippy::all`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#into_iter_on_ref
```

File: integration-tests/src/user/runtime_user.rs (hunk 1 above)
Comment (pmnoxx) [reply integration-tests/src/user/runtime_user.rs:172]: @mina86 Yes, it's not needed, it's strange that clippy doesn't warn about it.

Comment (pmnoxx) [reply integration-tests/src/user/runtime_user.rs:172]: I created an issue at https://github.com/rust-lang/rust-clippy/issues/8149

**Pull request merged**